### PR TITLE
Task/DES-1413 - Update project review process in publication pipeline

### PR DESCRIPTION
**Pull request by Strmiska**
**MERGES WITH TASK/DES-1358**
- Update the project review method
- Checking before and after the publication pipeline to prevent issues

### CODE REVIEW

File: designsafe/static/scripts/ng-designsafe/services/project-service.js [hunk 1]
```diff
@@ -120,88 +120,188 @@ export function ProjectService(httpi, $interpolate, $q, $state, $uibModal, Loggi
...
-                    });
-                }
             }
-
-        });
+            if (reports.length) {
+                checkRequirements(reports);
+            }
+        }
+        console.log(missingData);
```
Comment (jarosenb) [inline designsafe/static/scripts/ng-designsafe/services/project-service.js]: Do we want to log here?

File: designsafe/static/scripts/ng-designsafe/services/project-service.js [hunk 2]
```diff
@@ -120,88 +120,188 @@ export function ProjectService(httpi, $interpolate, $q, $state, $uibModal, Loggi
...
+            }
         };
 
-        let requiredSets = [];
-        let missingData = [];
-        let requiresFiles = ['analysis_set', 'report_set']; // checks if files are provided if included....
         if (project.value.projectType === 'experimental') {
-            requiredSets = ['modelconfig_set', 'sensorlist_set', 'event_set'];
+            /* 
+            Experimental Requirements:
```
Comment (fnets) [inline designsafe/static/scripts/ng-designsafe/services/project-service.js]: These comments spelling out the requirements for each project are super helpful

File: designsafe/static/scripts/ng-designsafe/services/project-service.js [hunk 3]
```diff
@@ -120,88 +120,188 @@ export function ProjectService(httpi, $interpolate, $q, $state, $uibModal, Loggi
...
+                + Sets must include categorized files
+            */
+            let requirements = ['model_config', 'sensor_list', 'event'];
+            let subentities = [].concat(
+                project.modelconfig_set || [],
+                project.sensorlist_set || [],
+                project.event_set || []
+            );
+            let experiments = selPrimEnts.filter(ent => ent.name.endsWith('experiment'));
+            // let reports = selPrimEnts.filter(ent => ent.name.endsWith('report'));
```
[the remark this answers is not in the document]
Comment (Strmiska) [reply designsafe/static/scripts/ng-designsafe/services/project-service.js]: I left these in place as a reminder that we may eventually want to publish reports for other project types in the same way that we publish reports for Field Research.

File: designsafe/static/scripts/ng-designsafe/services/project-service.js (hunk 2 above)
Comment (Strmiska) [reply designsafe/static/scripts/ng-designsafe/services/project-service.js]: Good to hear! :D

File: designsafe/static/scripts/ng-designsafe/services/project-service.js (hunk 1 above)
Comment (Strmiska) [reply designsafe/static/scripts/ng-designsafe/services/project-service.js]: Ack, no.

**Pull request merged**